### PR TITLE
MAID-1438

### PR DESCRIPTION
**Pull request by brian-js**
[Review on Reviewable](https://reviewable.io/reviews/maidsafe/routing/736)

### CODE REVIEW

**Comment by maidsafe-highfive** [comment]
r? @benjaminbollen

(maidsafe_highfive has picked a reviewer for you, use r? to override)

File: src/routing_core.rs [hunk 1]
```diff
@@ -73,6 +73,26 @@ pub enum State {
...
+    /// ConnectRequest sent by peer.
+    Request(::messages::ConnectRequest),
+    /// ConnectResponse in response to a ConnectRequest sent by peer.
+    Response(::messages::ConnectResponse),
+}
+
+/// UnknownConnection.
+#[derive(PartialEq, Eq, PartialOrd, Ord, Debug, Clone)]
+#[allow(unused)]
+pub struct UnknownConnection {
```
Comment (benjaminbollen) [inline src/routing_core.rs]: This is not the construction needed;

as in rfc-0011
RoutingCore {
  unknown_connections: ExpirationMap<::crust::Connection, Option<::direct_messages::Hello>>,

File: src/routing_node.rs [hunk 2]
```diff
@@ -138,6 +138,12 @@ impl RoutingNode {
                 Ok(Action::DropConnections(connections)) => {
                     self.drop_connections(connections);
                 },
+                Ok(Action::ExpectedConnection(expected_connection)) => {
```
Comment (benjaminbollen) [inline src/routing_node.rs]: Can the action include the actual verb "MatchExpectedConnection" and similarly for the other? Same for function names

**Comment by benjaminbollen** [comment]
looks good to me

**Pull request merged**